src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/LearnerDataRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.EmployerIncentives.Data.ApprenticeshipIncentives.Map;
+using SFA.DAS.EmployerIncentives.Data.Models;
+using SFA.DAS.EmployerIncentives.Domain.ApprenticeshipIncentives.ValueTypes;
+using System;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Data.ApprenticeshipIncentives
+{
+    public class LearnerDataRepository : ILearnerDataRepository
+    {
+        private readonly EmployerIncentivesDbContext _dbContext;
+
+        public LearnerDataRepository(EmployerIncentivesDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<Learner> GetByApprenticeshipIncentiveId(Guid apprenticeshipIncentiveId)
+        {
+            var learner = await _dbContext.Learners
+                            .FirstOrDefaultAsync(a => a.ApprenticeshipIncentiveId == apprenticeshipIncentiveId);
+
+            if (learner == null)
+            {
+                return null;
+            }
+
+            return learner.Map();
+        }
+
+        public async Task Save(Learner learner)
+        {
+            var updatedLearner = learner.Map();
+
+            var existingLearner = await _dbContext.Learners.FirstOrDefaultAsync(x => x.Id == updatedLearner.Id);
+
+            if (existingLearner != null)
+            {
+                _dbContext.Entry(existingLearner).CurrentValues.SetValues(updatedLearner);
+            }
+            else
+            {
+                _dbContext.Learners.Add(updatedLearner);
+            }
+
+            await _dbContext.SaveChangesAsync();
```

Comment:
This should be handled by the unit of work.

Reply:
Looks like the UoW is not wired up correctly in this project. As the function apps don't use the webapi pipline. 

I suggest we wait until we get the UoW implemented as part of the command handler (I believe that is WIP), at the moment I think all transactions in this project only update or add one record, so it's not problem until we start updating more than one record or send a NSB message

Reply:
Who's looking at the UoW?

Reply:
I thought you were, you mentioned adding the UoW into the Command handler layer and asked if we had any objections (I assume as a decorator) so we can use it correctly in the non API projects. Because we couldn't wire up EF and NSB transactions without the API pipeline

Reply:
not me, maybe @Stephek  🙄